File: flutter17_mobile/lib/models/news.dart
import 'package:json_annotation/json_annotation.dart';

part 'news.g.dart';

@JsonSerializable()
class News {
  int? id;
  String? title;
  String? content;
  DateTime? datePublished;

  News({
    this.id,
    this.title,
    this.content,
    this.datePublished,
  });

  factory News.fromJson(Map<String, dynamic> json) => _$NewsFromJson(json);

  Map<String, dynamic> toJson() => _$NewsToJson(this);
}

File: angular17_web/src/app/models/news/news.ts
export class News {
  id!: number;
  title!: string;
  content!: string;
  datePublished!: Date;

  constructor(data?: any) {
    if (!data) return;

    this.id = data.id;
    this.title = data.title;
    this.content = data.content;
    this.datePublished = data.datePublished ? new Date(data.datePublished) : undefined as any;
  }
}
